input/blog/2018-05-31-cake-v0.28.0-released.md

```diff
@@ -0,0 +1,196 @@
+    }
+    else
+    {
+        Information("Assemblies was NOT signed");
+    }
+});
+```
+
+### Setup context
+
+This release changed the script Setup context from a `ICakeContext` to `ISetupContext`. In addition to everything that `ICakeContext` offers, there is now some additional information provided:
+
+```csharp
+ICakeTaskInfo TargetTask { get; }
+IReadOnlyCollection<ICakeTaskInfo> TasksToExecute { get; }
+```
+
+`TargetTask` is the task that initiated the script and `TasksToExecute` are the tasks that will be executed based on the dependency graph of the target task. Access to which tasks actually will be executed. has been a greatly sought after feature, combined with the typed context makes it fairly straightforward to enable some advanced scenarios.
+
+```csharp
+Setup<MyBuildData>(setupContext =>
+{
+	var signAssemblies = BuildSystem.IsRunningOnAppVeyor
+							|| StringComparer.OrdinalIgnoreCase.Equals("Sign", setupContext.TargetTask.Name);
+
+	var isPublishBuild = setupContext
+                          .TasksToExecute
+                          .Where(task => StringComparer.OrdinalIgnoreCase.Equals("Publish", task.Name))
+                          .Any();
+
+    return new MyBuildData (
+    	configuration: Argument("configuration", "Release"),
+    	signAssemblies: signAssemblies,
+        isPublishBuild: isPublishBuild);
+});
+```
+
+In the above code we choose to sign assemblies if either running on AppVeyor or the script initiating target is `Sign`. And we'll treat build as a "publish" build if any of the depending tasks is named `Publish`.
+
+### Criteria messages
+
+Task `WithCriteria` can now have a message that's outputted when a task is skipped, this will make it clearer in the log why a specific task wasn't executed.
+
+This means by doing
+
+```csharp
+Task("Sign")
+    .WithCriteria<MyBuildData>(
+        (data, context) => data.SignAssemblies,
+        "Sign skipped because SignAssemblies is set to false."
+        );
+```
+
+the verbose output goes from
+
+```
+========================================
+Sign
+========================================
+Skipping task: Sign
+```
+
+to
+
+```
+========================================
+Sign
+========================================
+Skipping task: Sign skipped because SignAssemblies is set to false.
+```
+
+### Potentially breaking
+
+When we have to ship a breaking change in Cake, one piece of feedback that we have often heard is that it can be very frustrating to have a build suddenly fail, due to a new dependency on a higher version of `Cake.Core`. We have always advocated that all builds should be pinned to a specific version of Cake, so that a new breaking change does not affect your builds, however, we also understand that some people want to run their builds this way.
+
+In this release we've introduced the notion of a "Potentially breaking change", which is when we've refactored code we will not break the build but issue a warning but not break the build.
```

Comment:
`build but issue` -> `build, but issue`

also, the `but not break the build` at the end is a repeat, and isn't needed.